### ISSUE TITLE
Multiple Call to super in child class constructor does not give any error

### ISSUE THREAD

**Issue by zeeshanhanif** [Microsoft/TypeScript#5317]
A call to super multiple times in Child class' constructor does not give any error, is it expected behavior? Dose it initiate multiple parent objects?

```javascript

class E {
    name:string;
    constructor(theName: string,age:number) { 
        this.name = theName; 
        console.log("E constrcutor");
    }
    displayName():void {
        console.log(" Name = " + this.name);
    }
}
class F extends E {
    name:string;
    constructor(theName: string) {
        this.name = theName; 
        console.log("F constrcutor");
        super(theName,4);
        super("Hello",5); // No error here 
    }
}
let e: E = new E("E",1);
let f: F = new F("F");
f.displayName(); // return "Hello"

```

**Comment by wgebczyk** [comment]
you cannot use this before super call...
... and of course second super call makes no sense.

**Comment by zeeshanhanif** [comment]
Actually 'this' is working before super call, you can call super anywhere in in the constructor

**Comment by wgebczyk** [comment]
@zeeshanhanif OK, your statement has surprised me. My understanding is that this usage in constructor is prohibited under super call is fully resolved. Could you point me to specification of ES6 that allows that free this usage in derived class constructor with super somewhere in the middle?

**Comment by kitsonk** [comment]
Well, while not the ES6 Specification, MDN [shows clearly](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super) that is valid.  The specification makes [no assertions](http://www.ecma-international.org/ecma-262/6.0/#sec-super-keyword) to the positioning of the super call (or the number of times) as well.

**Comment by wgebczyk** [comment]
Can you reread Description section from your first link (MDN) the one after Syntax and before Example?
I'm reading it exactly as I've suggested:
```
When used in a constructor, the super keyword appears alone and must be used before the this keyword can be used. This keyword can also be used to call functions on a parent object.
```

**Comment by zeeshanhanif** [comment]
@wgebczyk 
Example in MDN also says calling 'this' before super will cause reference error.
But example I have provided above is working fine with typescript and does not give any error.

**Comment by wgebczyk** [comment]
OK, we misunderstood each other. You are saying that now it works, but it should not. :) I tried to say the same thing :)

**Comment by zeeshanhanif** [comment]
Yes, if we are following object oriented style like Java or C# then super can only be called in first line of constructor and only at once. But I'm not sure if current behavior in typescript is expected or not

**Comment by wgebczyk** [comment]
In [https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals] in Goals in point 6:
```Align with current and future ECMAScript proposals.```
So it seems that expected is to disallow this usage before super call.

**Comment by zeeshanhanif** [comment]
Ok, means we should expect that it will be restricted in future.

**Comment by mhegazy** [comment]
this should be handled by https://github.com/Microsoft/TypeScript/issues/4211
